temporal_sqlalchemy/core.py

```diff
@@ -0,0 +1,151 @@
+
+        activity_class = getattr(temporal_declaration, 'activity_class', None)
+        schema = getattr(temporal_declaration, 'schema', entity_table.schema)
+
+        clock_table = TemporalModel.build_clock_table(
+            entity_table,
+            cls.metadata,
+            schema,
+            activity_class
+        )
+        clock_properties = {
+            'entity': orm.relationship(
+                lambda: cls, backref=orm.backref('clock', lazy='dynamic')
+            ),
+            'entity_first_tick': orm.relationship(
+                lambda: cls,
+                backref=orm.backref(
+                    'first_tick',
+                    primaryjoin=sa.and_(
+                        clock_table.join(entity_table).onclause,
+                        clock_table.c.tick == 1
+                    ),
+                    innerjoin=True,
+                    uselist=False,  # single record
+                    viewonly=True  # view only
+                )
+            ),
+            'entity_latest_tick': orm.relationship(
+                lambda: cls,
+                backref=orm.backref(
+                    'latest_tick',
+                    primaryjoin=sa.and_(
+                        clock_table.join(entity_table).onclause,
+                        entity_table.c.vclock == clock_table.c.tick
+                    ),
+                    innerjoin=True,
+                    uselist=False,  # single record
+                    viewonly=True  # view only
+                )
+            ),
+            '__table__': clock_table
+        }  # used to construct a new clock model for this entity
+
+        if activity_class:
+            # create a relationship to the activity from the clock model
+            backref_name = '%s_clock' % entity_table.name
+            clock_properties['activity'] = \
+                orm.relationship(lambda: activity_class, backref=backref_name)
+
+        clock_model = clock.build_clock_class(cls.__name__,
+                                              cls.metadata,
+                                              clock_properties)
+
+        history_models = {
+            p: clock.build_history_class(cls, p, schema)
+            for p in tracked_props
+        }
+
+        cls.temporal_options = bases.ClockedOption(
+            temporal_props=tracked_props,
+            history_models=history_models,
+            clock_model=clock_model,
+            activity_cls=activity_class
+        )
+
+        event.listen(cls, 'init', TemporalModel.init_clock)
+
+    @staticmethod
+    def init_clock(clocked: 'TemporalModel', args, kwargs):
+        kwargs.update({'vclock': 1})
+        inital_tick = clocked.temporal_options.clock_model(
+            tick=kwargs['vclock'],
+            entity=clocked,
+        )
+
+        if 'activity' in kwargs:
+            inital_tick.activity = kwargs.pop('activity')
+
+    @declarative.declared_attr
+    def __mapper_cls__(cls):
```

Comment:
Is this what does the magic? SQLA provides a hook to look for metaclasses? If so can you add a comment explaining how it works a bit?